api/agent/data_access.go

```diff
@@ -82,6 +84,10 @@ func (da *CachedDataAccess) GetApp(ctx context.Context, appName string) (*models
 	return app.(*models.App), nil
 }
 
+func (da *CachedDataAccess) GetApp(ctx context.Context, app *models.App) (*models.App, error) {
+	return da.DataAccess.GetApp(ctx, app)
```

Comment:
can have a separate cache for by name. i agree that this method should stay what it was instead of taking a partially filled in `*models.App`, it's a little weird